SynthDet/Assets/Scripts/Randomizers/BackgroundObjectPlacementRandomizer.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Perception.Randomization.Parameters;
+using UnityEngine.Perception.Randomization.Randomizers;
+using UnityEngine.Perception.Randomization.Randomizers.Utilities;
+using UnityEngine.Perception.Randomization.Samplers;
+
+namespace SynthDet.Randomizers
+{
+    /// <summary>
+    /// Creates multiple layers of evenly distributed but randomly placed objects
+    /// </summary>
+    [Serializable]
+    [AddRandomizerMenu("SynthDet/My Background Object Placement Randomizer")]
+    public class BackgroundObjectPlacementRandomizer : Randomizer
+    {
+        /// <summary>
+        /// The Z offset component applied to all generated background layers
+        /// </summary>
+        public float depth;
+
+        /// <summary>
+        /// The number of background layers to generate
+        /// </summary>
+        public int layerCount = 2;
+
+        /// <summary>
+        /// The minimum distance between placed background objects
+        /// </summary>
+        public float separationDistance = 0.5f;
+
+        /// <summary>
+        /// The 2D size of the generated background layers
+        /// </summary>
+        public Vector2 placementArea = new Vector2(6f, 6f);
+
+        /// <summary>
+        /// A categorical parameter for sampling random prefabs to place
```

Comment:
Is it still called "categorical parameter"?

Reply:
yes